merci/evaluate.py

```diff
@@ -259,9 +259,44 @@ def evaluate(self) -> float:
             attr_i = np.where(X_test[self.sensitive_attribute_idx] == i)
             y_i = y_test[attr_i]
             y_pred_i = np.array(self.model.predict(X_test[attr_i]))
-            
+
             tp.append(np.sum((y_pred_i == 1) & (y_i == 1)) / np.sum(y_i == 1))
             fp.append(np.sum((y_pred_i == 1) & (y_i == 0)) / np.sum(y_i == 0))
 
         equalized_odds = (np.abs(tp[0] - tp[1]) + np.abs(fp[0] - fp[1])) / 2
         return equalized_odds
+
+
+class CalibrationEvaluator(ModelEvaluator):
+    """
+    Evaluate expected calibration error
+    """
+
+    def __init__(
+        self,
+        model: merci.types.Classifier,
+        train_dataset: merci.types.Dataset,
+        test_dataset: merci.types.Dataset,
+        seed: int = 0,
+    ) -> None:
+        """
+        Initialize the transductive model evaluator.
+
+        :param model: Classification model to evaluate
+        :param train_dataset: A tuple containing input and target data for training
+        :param test_dataset: A tuple containing input and target data for testing
+        """
+        super().__init__(model, train_dataset, test_dataset)
+
+    def evaluate(self) -> float:
+        """
+        Evaluate the model using the provided dataset.
+
+        :return: The expected calibration error of the model on the dataset
+        """
+        x_train, y_train = self.train_dataset
+        x_test, y_test = self.test_dataset
+        model = self.model.fit(x_train, y_train)
+        preds = model.predict_proba(x_test)[:, 1]
```

Comment:
`merci.types.Classifer` Protocol does not have a `predict_proba` - maybe you could create a new protocol class that specifically defines a classifier that specifically returns a probability distribution and use it for a type annotation (I could also use it in the TransductiveEvaluator

Reply:
(this way it will be clearer for the user of the library how to make a wrapper for the tested model - i.e. whether the predictor should return classes or probabilities)

Reply:
done